### PR TITLE
feat: gas fee price

### PR DESCRIPTION
**Pull request by 0xtiti**
closes SUP-68

### CODE REVIEW

File: src/utils/calculateGasFee.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { fetchTokenPrice } from '.';
```
[the remark this answers is not in the document]
Comment (0xtiti) [reply src/utils/calculateGasFee.ts]: [b90944c](https://github.com/defi-wonderland/superchain-bridge-ui/pull/69/commits/b90944c49adfc52087a2067ef605e30c9d49669b)

File: src/containers/Modals/Review.tsx [hunk 2]
```diff
@@ -21,9 +30,28 @@ export const ReviewModal = () => {
...
   const { executeTransaction } = useTransactions();
   const [copiedStates, copy] = useCopyToClipboard();
+  const { fromPublicClient } = useCustomClient();
+  const [gasFee, setGasFee] = useState<string>('');
 
+  const gasUsed = gasFees[transactionType];
   const totalAmount = amount || mint || value;
 
+  const fetchGasPrice = useCallback(async () => {
+    const gasPrice = await fromPublicClient.getGasPrice();
```
Comment (0xArdy) [inline src/containers/Modals/Review.tsx:40]:
```suggestion
  const { customClient } = useCustomClient();
  const [gasFee, setGasFee] = useState<string>('');

  const gasUsed = gasFees[transactionType];
  const totalAmount = amount || mint || value;

  const fetchGasPrice = useCallback(async () => {
    const gasPrice = await customClient.from.public.getGasPrice();
```

File: src/providers/CustomClientProvider.tsx [hunk 3]
```diff
@@ -8,9 +8,14 @@ import { getFromContracts, getToContracts } from '~/utils';
...
 import { alchemyUrls } from '~/data';
 
+interface PublicClient {
+  getGasPrice: () => Promise<bigint>;
+}
+
 type ContextType = {
   customClient: CustomClients;
   logsClient: CustomClients;
+  fromPublicClient: PublicClient;
```
Comment (0xArdy) [inline src/providers/CustomClientProvider.tsx:18]: I think these changes are unnecessary, why dont you use `customClient.from.public`?

**Review by 0xArdy — APPROVED** [review]
lgtm!

**Review by agusduha — APPROVED** [review]

**Pull request merged**